### PR TITLE
[UI] Family test selection and consent screen (EXPOSUREAPP-12316, EXPOSUREAPP-12317)

### PR DESCRIPTION
**Pull request by AlexanderAlferov**
### Whats new

Test type selection screen

Screenshot 2022-03-17 at 18 12 56

##Family member consent consent

Screenshot 2022-03-17 at 20 04 16

### Story
[Story](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5895)
[Test type selection](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12316)
[Family member](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12317)

### How to test
1) scan any test
2) check test type selection fragment
3) press "Für andere" button
4) check family consent fragment
5) check that with empty name confirmation button is not enabled
6) type any name
7) check confirmation button is now enabled

### ToDo

- [x] Strings
- [x] New icons and illustration
- [x] Layouts
- [x] Wiring up

### CODE REVIEW

File: Corona-Warn-App/src/main/res/values-de/family_tests_strings.xml [hunk 1]
```diff
@@ -1,4 +1,32 @@
...
+    <!-- XTXT: Test registration selection: subtitle -->
+    <string name="family_consent_subtitle">"Testergebnis für Familienmitglieder abrufen"</string>
+    <!-- XTXT: Test registration selection: person name label -->
+    <string name="family_consent_person_name_label">"Name der Person, die getestet wurde:"</string>
+    <!-- XTXT: Test registration selection: person name tooltip -->
+    <string name="family_consent_person_name_tooltip">"Name"</string>
+    <!-- XTXT: Test registration selection: test type -->
+    <string name="family_consent_test_type">"Sie registrieren einien Test für ein Familienmitglied. Sollte es ihr eigener sein gehen Sie bitte zurück und wählen das aus. Nur so können Sie Ihre Begegnungen warnen!"</string>
+    <!-- XTXT: Test registration selection: test type -->
+    <string name="family_consent_test_result">"Wenn Ihr Testergebnis negativ ist, können Sie dies in Form eines offiziellen COVID-Testzertifikats bestätigen lassen. Fordern Sie hierfür in den folgenden Schritten das Testzertifikat an."</string>
```
Comment (Ein-Tim) [inline Corona-Warn-App/src/main/res/values-de/family_tests_strings.xml:28]:
```suggestion
    <string name="family_consent_test_result">"Wenn das Testergebnis negativ ist, können Sie dies in Form eines offiziellen COVID-Testzertifikats bestätigen lassen. Fordern Sie hierfür in den folgenden Schritten das Testzertifikat an."</string>
```

File: Corona-Warn-App/src/main/res/values-de/legal_strings.xml [hunk 2]
```diff
@@ -229,4 +229,14 @@
...
     <!-- XTXT: DCC Reissuance legal second bullet point-->
     <string name="dcc_reissuance_legal_two" translatable="false">Das RKI überprüft das Impfzertifikat und stellt ein aktualisiertes Zertifikat aus und übermittelt dieses an Ihre App.</string>
+
+    <!-- Family Consent Screen -->
+    <!-- XTXT: Family Consent Screen: Legal: Title -->
+    <string name="family_consent_legal_title" translatable="false">"Ihr Einverständnis"</string>
+    <!-- XTXT: Family Consent Screen: Legal: Body -->
+    <string name="family_consent_legal_body" translatable="false">"Durch Antippen von „Einverstanden“ willigen Sie in folgende Schritte ein:"</string>
+    <!-- XTXT: Family Consent Screen: Legal: Point 1 -->
+    <string name="family_consent_legal_point_1" translatable="false">"<b>Die App ruft Ihr Testergebnis ab.</b>\nWenn Sie es sich später anders überlegen, können Sie den Test in der App löschen."</string>
```
Comment (Ein-Tim) [inline Corona-Warn-App/src/main/res/values-de/legal_strings.xml:239]:
```suggestion
    <string name="family_consent_legal_point_1" translatable="false">"<b>Die App ruft das Testergebnis ab.</b>\nWenn Sie es sich später anders überlegen, können Sie den Test in der App löschen."</string>
```

**Review by Ein-Tim — CHANGES_REQUESTED** [review]
Bitte nicht "Ihr Testergebnis" nutzen, wenn es um das Testergebnis eines Familienmitgliedes geht. Das wird sonst verwirrend.

**Comment by AlexanderAlferov** [comment]
> Bitte nicht "Ihr Testergebnis" nutzen, wenn es um das Testergebnis eines Familienmitgliedes geht. Das wird sonst verwirrend.

@KathrinLuecke FYI 👀

File: Corona-Warn-App/src/main/res/values-de/family_tests_strings.xml [hunk 3]
```diff
@@ -1,4 +1,32 @@
...
+    <!-- XHED: Test registration selection: header -->
+    <string name="family_consent_header">"Ihr Einverständnis"</string>
+    <!-- XTXT: Test registration selection: subtitle -->
+    <string name="family_consent_subtitle">"Testergebnis für Familienmitglieder abrufen"</string>
+    <!-- XTXT: Test registration selection: person name label -->
+    <string name="family_consent_person_name_label">"Name der Person, die getestet wurde:"</string>
+    <!-- XTXT: Test registration selection: person name tooltip -->
+    <string name="family_consent_person_name_tooltip">"Name"</string>
+    <!-- XTXT: Test registration selection: test type -->
+    <string name="family_consent_test_type">"Sie registrieren einien Test für ein Familienmitglied. Sollte es ihr eigener sein gehen Sie bitte zurück und wählen das aus. Nur so können Sie Ihre Begegnungen warnen!"</string>
```
Comment (Ein-Tim) [inline Corona-Warn-App/src/main/res/values-de/family_tests_strings.xml:26]:
```suggestion
    <string name="family_consent_test_type">"Sie registrieren einien Test für ein Familienmitglied. Sollte es ihr eigener sein gehen Sie bitte zurück und wählen das aus. Nur so können Sie Ihre Kontakte im Falle eines positiven Testergebnisses über die App warnen!"</string>
```

**Review by Ein-Tim — COMMENTED** [review]
Make it clearer what is meant here.

**Comment by KathrinLuecke** [comment]
It looks like the screens for Android are not up-to-date in Figma. They are different from the iOS screens with regards to the wording. I will aks Anna-Maria to update them, then the texts here have to be adapted.

**Comment by KathrinLuecke** [comment]
iOS:
image

**Review by KathrinLuecke — DISMISSED** [review]
UA approved

**Review by CV113 — DISMISSED** [review]
UA approved

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt [hunk 4]
```diff
@@ -1,6 +1,38 @@
...
+        binding.family.setOnClickListener {
+            doNavigate(
+                TestRegistrationSelectionFragmentDirections
+                    .actionTestRegistrationSelectionFragmentToFamilyTestConsentFragment(
+                        coronaTestQrCode = navArgs.coronaTestQrCode
+                    )
+            )
+        }
+    }
+}
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt:38]: Please rename `CoronaTestQRCode.CategoryType.OWN` to `CoronaTestQRCode.CategoryType.PERSONAL` for consistency?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt [hunk 5]
```diff
@@ -1,6 +1,38 @@
...
+    private val binding: FragmentTestRegistrationSelectionBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.toolbar.setNavigationOnClickListener { popBackStack() }
+        binding.person.setOnClickListener {
+            doNavigate(
+                TestRegistrationSelectionFragmentDirections
+                    .actionTestRegistrationSelectionFragmentToSubmissionConsentFragment(
+                        coronaTestQrCode = navArgs.coronaTestQrCode
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt:25]: you need to create a copy of `coronaTestQrCode` and pass the right type based on user selection

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt [hunk 6]
```diff
@@ -1,6 +1,38 @@
...
+                    )
+            )
+        }
+        binding.family.setOnClickListener {
+            doNavigate(
+                TestRegistrationSelectionFragmentDirections
+                    .actionTestRegistrationSelectionFragmentToFamilyTestConsentFragment(
+                        coronaTestQrCode = navArgs.coronaTestQrCode
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt:33]: you need to create a copy of `coronaTestQrCode` and pass the right type based on user selection

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt [hunk 7]
```diff
@@ -1,6 +1,38 @@
...
+    private val navArgs by navArgs<TestRegistrationSelectionFragmentArgs>()
+    private val binding: FragmentTestRegistrationSelectionBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        binding.toolbar.setNavigationOnClickListener { popBackStack() }
+        binding.person.setOnClickListener {
+            doNavigate(
+                TestRegistrationSelectionFragmentDirections
+                    .actionTestRegistrationSelectionFragmentToSubmissionConsentFragment(
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt:24]: that is not always the case , if there a test of the same type , you need to open the duplicate warning screen , see Figma

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+    fun onNavigateBack() {
+        routeToScreen.postValue(FamilyTestConsentNavigationEvents.NavigateBack)
+    }
+
+    fun onConsentButtonClick() = launch {
+        FamilyTestConsentNavigationEvents.NavigateToCertificateRequest(
+            coronaTestQRCode = coronaTestQRCode,
+            consentGiven = true,
+            allowReplacement = true,
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/consent/FamilyTestConsentViewModel.kt:45]: should be false , family test are allowing many tests we don't have replacement for them

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt [hunk 9]
```diff
@@ -1,6 +1,72 @@
...
+            factory as TestRegistrationSelectionViewModel.Factory
+            factory.create(navArgs.coronaTestQrCode)
+        }
+    )
+    private val binding: FragmentTestRegistrationSelectionBinding by viewBinding()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        viewModel.routeToScreen.observe2(this) {
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/familytest/ui/selection/TestRegistrationSelectionFragment.kt:34]: This screen stays in the navigation back stack even after the test is registered, I think when we navigate from here, it should be piped up from the stack

**Review by mtwalli — COMMENTED** [review]
Great job ,slick UI

**Review by mtwalli — DISMISSED** [review]
🚀🚀🚀🚀🚀

**Review by SamuraiKek — APPROVED** [review]
Let's go! 🚀

**Pull request merged**